envoy-control-core/src/main/resources/lua/ingress_client_name_header.lua

```diff
@@ -0,0 +1,25 @@
+function envoy_on_request(handle)
+    local streamInfo = handle:streamInfo()
+    local trusted_header = handle:metadata():get("trusted_client_identity_header") or ""
+    if trusted_header == "" then
+        return
+    end
+
+    if handle:headers():get(trusted_header) ~= nil then
+        handle:headers():remove(trusted_header)
+    end
+
+    if handle:connection():ssl() and streamInfo:downstreamSslConnection() then
+        local uriSanPeerCertificate = handle:streamInfo():downstreamSslConnection():uriSanPeerCertificate()
+        if uriSanPeerCertificate ~= nil and next(uriSanPeerCertificate) ~= nil then
```

Comment:
Why `next`? What if `uriSanPeerCertificate` only has one entry?

Reply:
`next` returns nil if the table is empty. It doesn't advance an iterator or something like that.